gematria/granite/python/token_graph_builder_model_test.py

```diff
@@ -577,6 +578,70 @@ def test_train_seq2seq_returning_error_on_unknown_token(self):
       model.schedule_batch(self.blocks_with_throughput)
 
 
+class TokenGraphBuilderModelEsotericBlocksTest(
+    parameterized.TestCase, model_test.TestCase
+):
+  """Test that we can train on esoteric blocks.
+
+  Test that we can handle blocks containing rather esoteric instruction
+  features, such as instruction prefixes, correctly during training.
+  """
+
+  def setUp(self):
+    def _block_filter(
+        basic_block: throughput_pb2.BasicBlockWithThroughputProto,
+    ):
+      prefixes = basic_block.basic_block.canonicalized_instructions[0].prefixes
+      if len(prefixes) == 0:
+        return False
+      return (
+          basic_block.basic_block.canonicalized_instructions[0].prefixes[0]
+          == 'REP'
+      )
```

Comment:
Let's make this either a module-level function (it doesn't use the `self` or the class in any way).